lib/ace/mode/yaml_highlight_rules.js

```diff
@@ -73,6 +73,22 @@ var YamlHighlightRules = function() {
             }, {
                 token : "string", // multi line string start
                 regex : '[|>][-+\\d\\s]*$',
+                onMatch: function(val, state, stack, line) {
+                    var indent = new RegExp(/^\s*(?:[\-?](?:$|\s))?/).exec(line)[0];
```

Comment:
why is `new Regexp` call needed? also `$` should not be possible, since this gets called only after `|`
I think you can change this this to `/^s*(?:[-?]\s)?/.exec(line)[0];`